docs/md/dev-process.md

```diff
@@ -77,7 +72,7 @@ For the release process, please refer to the [release guide](../dev-release/)
     shown below so JIRA could pick up the branch and link it
```

Comment:
Off topic but I thought I did this correctly on my PR and did not see the link take place.  Do we know if this still works?

Reply:
It should be in the "development" section of the ticket on the right-hand-side?

Reply:
interesting, I never followed this naming convention on my branches, but as long as I use `[GH-xxx]` in my PR title, Jira picks it up.